pkg/aggregateeventrecorder/aggregateeventrecorder.go

```diff
@@ -0,0 +1,172 @@
+package aggregateeventrecorder
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/filecoin-project/lassie/pkg/events"
+	"github.com/filecoin-project/lassie/pkg/types"
+	"github.com/ipfs/go-log/v2"
+)
+
+var logging = log.Logger("aggregateeventrecorder")
+
+const httpTimeout = 5 * time.Second
+const parallelPosters = 5
+
+type tempData struct {
+	startTime time.Time
+	ttfb      float64
+}
+
+type AggregatedEvent struct {
+	InstanceID        string    `json:"instanceId"`        // The ID of the Lassie instance generating the event
+	RetrievalID       string    `json:"retrievalId"`       // The unique ID of the retrieval
+	StorageProviderID string    `json:"storageProviderId"` // The ID of the storage provider that served the retrieval content
+	TimeToFirstByte   float64   `json:"timeToFirstByte"`   // The time it took to receive the first byte in seconds
+	Bandwidth         float64   `json:"bandwidth"`         // The bandwidth of the retrieval in bytes per second
```

Comment:
would be nice to avoid floats for these if possible—ttfb could be done in milliseconds, or even smaller if need be, bandwidth surely doesn't need to be a float for _bytes_ per second, does it?

Reply:
Yeah, I didn't like it when I wrote it, but I wanted bytes per second at the time. I could throw it over the wire as a string that truncates everything past two decimals? We don't need crazy precision.